### ISSUE TITLE
batchByName fails on empty parameter lists

### ISSUE THREAD

**Issue by imre-kerr** [scalikejdbc/scalikejdbc#481]
ScalikeJDBC version 2.3.4

If the query actually includes named placeholders (I notice the test in #467 doesn't), `batchByName` fails with a `java.lang.IllegalArgumentException - Binding parameter is missing. ('name)`

**Comment by cppexpert** [comment]
What is the use-case for passing empty parameter list ?

**Comment by imre-kerr** [comment]
I get a (potentially empty) list of Scala objects to be inserted in the DB:
```
val batchParams = objects.map(o => Seq(
  'foo -> o.foo,
  'bar -> o.bar
))
```
then, this fails:
```
sql"INSERT INTO some_table (foo, bar) VALUES ({foo}, {bar})".batchByName(batchParams: _*).apply()
```

**Comment by cppexpert** [comment]
I see. Yeah, it would be interesting to see what @seratch says. From one hand one can argue that method shouldn't fail but o the other hand doing nothing on empty list is unintended side-effect.

**Comment by seratch** [comment]
@imre-kerr Thank you for your feedback. I just fixed this issue by #483. If you have time, take a look at it.
